### ISSUE TITLE
absolute_path_linter() fails for unmarked UTF-8

### ISSUE THREAD

**Issue by bastistician** [jimhester/lintr#3046]
Reproducible example, assuming a UTF-8 session:

``` r
library("lintr")

tf <- tempfile()
writeLines('x <- c("\u00e4")', tf, useBytes = TRUE)

lint(tf, linters = absolute_path_linter())                   # OK
#> ℹ No lints found.

text_UTF8 <- readLines(tf, encoding = "UTF-8")
lint(text = text_UTF8, linters = absolute_path_linter())     # OK
#> ℹ No lints found.

text_native <- readLines(tf)  # should be fine in a UTF-8 session
lint(text = text_native, linters = absolute_path_linter())
#> Error in `lint_impl_()`:
#> ! Linter `linter()` failed in '/tmp/RtmpTmjsLz/file185f638e6615c':
#> Caused by error in `parse()`:
#> ! <text>:1:5: Unerwartete(s) ')'
#> 1: "ä")
#>         ^
```

Possibly related to #2836.

**Comment by MichaelChirico** [comment]
IIUC this _only_ applies when passing in _via_ `text=` right?

**Comment by bastistician** [comment]
Seems so. In the first two examples that are OK, the `lint`-internal `lines <- get_lines(filename, text)` has a marked UTF-8 encoding, in the last case `Encoding(lines)` is "unknown", which seems to cause problems further down the line even if UTF-8 is the native encoding.

BTW, the error message ``Linter `linter()` failed ...`` probably intended to substitute `linter` by its value.
